### PR TITLE
Refactor text measurement & wrapping logic away from TextNode

### PR DESCRIPTION
**Pull request by iMoses**
WIP - need to fix sunburst label not disappearing

### CODE REVIEW

File: packages/ag-charts-community/src/chart/axis/__image_snapshots__/axis-examples-test-ts-axis-examples-auto-wrap-axis-labels-cases-for-stacked-column-chart-example-it-should-render-to-canvas-as-expected-1-snap.png
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/axis/__image_snapshots__/axis-examples-test-ts-axis-examples-auto-wrap-axis-labels-cases-for-stacked-column-chart-example-it-should-render-to-canvas-as-expected-1-snap.png:1]: So, it looks like there is a bias for breaking a line early vs. the previous implementation? Is that intentional?

File: packages/ag-charts-community/src/util/textMeasurer.ts [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+                const lastSpaceIndex = beforeLast.lastIndexOf(' ');
+                if (lastSpaceIndex !== -1 && !lastLine.includes(' ')) {
+                    const lastWord = beforeLast.slice(lastSpaceIndex + 1);
+                    if (measurer.textWidth(lastLine + lastWord) <= options.maxWidth) {
+                        result[length - 2] = beforeLast.slice(0, lastSpaceIndex);
+                        result[length - 1] = lastWord + ' ' + lastLine;
+                    }
+                }
+            }
+        }
```
Comment (alantreadway) [inline packages/ag-charts-community/src/util/textMeasurer.ts]: Nit: Any way to flatten the control flow out a bit here?

File: packages/ag-charts-community/src/util/textMeasurer.ts [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+            for (let i = 0, estimatedWidth = 0, lastSpaceIndex = 0; i < line.length; i++) {
+                const char = line.charAt(i);
+
+                estimatedWidth += measurer.textWidth(char);
+
+                if (char === ' ') {
+                    lastSpaceIndex = i;
+                }
+
+                if (estimatedWidth > options.maxWidth) {
```
Comment (alantreadway) [inline packages/ag-charts-community/src/util/textMeasurer.ts:115]: Nit: Would it improve readability if we decomposed this block a bit (maybe in a distinct function)?

File: packages/ag-charts-community/src/scene/shape/__image_snapshots__/text-test-ts-text-rendering-should-wrap-and-render-as-expected-1-snap.png
Comment (alantreadway) [inline packages/ag-charts-community/src/scene/shape/__image_snapshots__/text-test-ts-text-rendering-should-wrap-and-render-as-expected-1-snap.png:1]: This definitely looks like it's working better - nice work!

File: packages/ag-charts-enterprise/src/series/treemap/__image_snapshots__/treemap-series-test-ts-treemap-series-series-highlighting-should-render-a-complex-chart-1-snap.png
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/series/treemap/__image_snapshots__/treemap-series-test-ts-treemap-series-series-highlighting-should-render-a-complex-chart-1-snap.png:1]: The wrapping change here looks like a bug (confirmed with @jacobp100 that this isn't expected)

File: packages/ag-charts-community/src/chart/axis/__image_snapshots__/axis-examples-test-ts-axis-examples-auto-wrap-axis-labels-cases-for-bar-chart-example-it-should-render-to-canvas-as-expected-1-snap.png
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/axis/__image_snapshots__/axis-examples-test-ts-axis-examples-auto-wrap-axis-labels-cases-for-bar-chart-example-it-should-render-to-canvas-as-expected-1-snap.png:1]: The breaking of `Sales & customer services` looks different (not sure if it's right or wrong tbh)?

**Review by alantreadway — APPROVED** [review]
LGTM

File: packages/ag-charts-community/src/chart/axis/__image_snapshots__/axis-examples-test-ts-axis-examples-auto-wrap-axis-labels-cases-for-bar-chart-example-it-should-render-to-canvas-as-expected-1-snap.png
Comment (iMoses) [reply packages/ag-charts-community/src/chart/axis/__image_snapshots__/axis-examples-test-ts-axis-examples-auto-wrap-axis-labels-cases-for-bar-chart-example-it-should-render-to-canvas-as-expected-1-snap.png:1]: it's open to interpretation 😅

**Pull request merged**